### PR TITLE
Refactor reconciliation controller

### PR DESCRIPTION
**Pull request by jveski**
More details in PR comments.

### CODE REVIEW

File: internal/controllers/reconciliation/controller.go [hunk 1]
```diff
@@ -142,37 +131,20 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("getting resource slice: %w", err))
 	}
 	var ready *metav1.Time
-	if len(slice.Status.Resources) > req.Manifest.Index {
```
Comment (jveski) [inline internal/controllers/reconciliation/controller.go:145]: Push more of the responsibility out of this package

File: internal/controllers/reconciliation/controller.go [hunk 2]
```diff
@@ -142,37 +131,20 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
...
 	}
 
-	// Store the results
 	if modified {
 		return ctrl.Result{Requeue: true}, nil
 	}
+
+	// Store the results
 	deleted := current == nil || current.GetDeletionTimestamp() != nil
-	c.resourceClient.PatchStatusAsync(ctx, &req.Manifest, func(rs *apiv1.ResourceState) *apiv1.ResourceState {
```
Comment (jveski) [inline internal/controllers/reconciliation/controller.go:166]: No need for this lambda to be inline

File: internal/discovery/cache.go [hunk 3]
```diff
@@ -12,16 +12,16 @@ import (
 	"k8s.io/kube-openapi/pkg/util/proto"
 )
 
-// discoveryCache is useful to prevent excessive QPS to the discovery APIs while
+// Cache is useful to prevent excessive QPS to the discovery APIs while
 // still allowing dynamic refresh of the openapi spec on cache misses.
-type discoveryCache struct {
+type Cache struct {
```
Comment (jveski) [inline internal/discovery/cache.go:17]: The discovery cache has very limited surface area with the controller, so it's a good candidate to have its own package.

**Review by AYM1607 — APPROVED** [review]
🌟

**Pull request merged**